tests/cursor/cursor-session-001.phpt

```diff
@@ -26,7 +26,7 @@ $iterator = new IteratorIterator($cursor);
 $iterator->rewind();
 $iterator->next();
 
-printf("Cursor ID is zero: %s\n", (string) $cursor->getId() === '0' ? 'yes' : 'no');
+printf("Cursor ID is zero: %s\n", $cursor->getId(true) == 0 ? 'yes' : 'no');
```

Comment:
Note that if `$cursor->getId(true)` return `null`, this condition `null == 0` would be `true`. Should you test the type of the returned value?

Reply:
Good point. I'll double check this, as it may have been related to the casting issues fixed in #1617.

Reply:
TL;DR: this is not an issue, because `getId()` would never return `null`, and even if it did the comparison logic we rely on for `Int64` instances handles this correctly.

The dirty details: before the change to `getId(true)`, we'd always get a `CursorId` instance, which does not implement cast or comparison handlers, so the only sensible way to assert on it was by casting it to a string and doing a typesafe comparison.

Now that we return an `Int64` instance, we can rely on PHP's internal logic for checking equality (not identity) in objects. Note that when using `===`, there's an [early exit in `zend_is_identical`](https://github.com/php/php-src/blob/fad899e5662d0a929d8462dd8239b0489dd9b53f/Zend/zend_operators.c#L2389-L2391) if the types do not match. So `$int64 === 0` will always evaluate to `false` based on the different types. `==` on the other hand calls `zend_is_equal`, which in turn calls `zend_compare`, which relies on an object's `compare` handler, and thus defers to the logic we implemented for comparing `Int64` instances with other values (and importantly does not include a cast to avoid issues on 32-bit platforms).

I did not add a specific test for the case you mentioned, as `zend_compare` does not call into the `compare` handler when comparing an object to `null`, instead [handling it specifically to ensure correct behaviour](https://github.com/php/php-src/blob/fad899e5662d0a929d8462dd8239b0489dd9b53f/Zend/zend_operators.c#L2317-L2321).